src/coreclr/tools/Common/JitInterface/UnboxingMethodDesc.cs

```diff
@@ -74,6 +76,13 @@ protected override int CompareToImpl(MethodDesc other, TypeSystemComparer compar
         {
             throw new NotImplementedException();
         }
+
+        protected override int ComputeHashCode()
+        {
+            throw new NotSupportedException("This method may not be stored  as it is expected to only be used transiently in the JIT");
```

Comment:
Nice, thanks for adding the check modulo the nit regarding double blank before 'as' ;-).